modules/test-utils/src/scene-renderer.js

```diff
@@ -112,7 +122,28 @@ export default class SceneRenderer {
     return promise;
   }
 
+  _getNextSceneIndex(sceneIndex) {
+    let nextSceneIndex = sceneIndex;
+    while (nextSceneIndex < this.scenes.length) {
+      const scene = this.scenes[nextSceneIndex];
+      let skipScene = false;
+      if (scene.gpuVendorSkipList) {
+        skipScene = scene.gpuVendorSkipList.reduce((skip, vendor) => {
+          skip = skip || vendor.startsWith(this.gpuVendor);
+          return skip;
+        });
+      }
+      if (!skipScene) {
+        break;
+      }
+      console.log(`Skipping render test ${scene.name} for ${this.gpuVendor} GPU`); // eslint-disable-line
+      nextSceneIndex++;
+    }
+    return nextSceneIndex;
+  }
+
   _renderNextScene() {
+    this.sceneIndex = this._getNextSceneIndex(this.sceneIndex);
     const scene = this.scenes[this.sceneIndex];
```

Comment:
Should we move the following lines of code into _getNextSceneIndex? Seems like they also do a related thing.

Reply:
Some of these lines are for setting other state.